sdk-java/src/test/java/ly/count/sdk/java/internal/ModuleDeviceIdTests.java

```diff
@@ -422,6 +423,164 @@ public void acquireId_customId() {
+        //if device id is not merged, then device id change request should not exist
+        validateDeviceIdWithoutMergeChange(3, TestUtils.keysValues[0]); // additional 1 session end 1 session begin, no events because no events exist
+    }
+
+    /**
+     * "setID" first sdk generated then developer supplied
+     * Validating that first call acts like "changeWithMerge" and second call acts like "changeWithoutMerge"
+     * SDK must generate an id first, then should change with developer supplied two times
+     * <br>
+     * - Init the SDK with no custom id
+     * - Setup session, view and event to simulate a device id change
+     * - Validate that session is begun
+     * - Validate that device id is sdk generated
+     * - Set new device id TestUtils.DEVICE_ID
+     * - Validate that EQ size is not changed because it is merge request
+     * - Validate that callback is called once which is "deviceIdChanged" internal callback
+     * - Validate that there are 2 requests in the event queue (first is begun session second one is device id change request)
+     * - Flush the request queue with old device id
+     * - Change device id to TestUtils.DEVICE_ID + "1"
+     * - Validate that EQ is flushed because it is not a merge request
+     * - Validate that callback is called twice
+     * - Validate that there are 3 requests in the request queue, first began session, 1 events, 1 end session
+     */
+    @Test
+    public void setID_changeWithMerge_then_withoutMerge() throws InterruptedException {
+        TestUtils.AtomicString deviceID = new TestUtils.AtomicString(TestUtils.DEVICE_ID);
+        AtomicBoolean withMerge = new AtomicBoolean(false);
+        AtomicInteger callCount = initDummyModuleForDeviceIdChangedCallback(deviceID, withMerge, DeviceIdType.DEVELOPER_SUPPLIED);
+        Countly.instance().init(TestUtils.getConfigDeviceId(null)); // to create sdk generated device id
+        setupView_Event_Session();
+        Assert.assertEquals(1, TestUtils.getCurrentRQ().length); // began session request
+        // validate began session request with generated id
+        validateBeganSessionRequest();
+        validateDeviceIdIsSdkGenerated(); // validate device id generated by the sdk
+
+        String oldDeviceId = Countly.instance().deviceId().getID();
+        Assert.assertEquals(0, callCount.get()); // validate "deviceIdChanged" callback not called
+
+        Assert.assertEquals(2, TestUtils.getCurrentEQ().size()); // size should not change because it is merge request
+        Countly.instance().deviceId().setID(TestUtils.DEVICE_ID); // TestUtils.DEVICE_ID
+        Assert.assertEquals(2, TestUtils.getCurrentEQ().size()); // size should not change because it is merge request
+
+        Assert.assertEquals(1, callCount.get());
+        validateDeviceIdWithMergeChange(oldDeviceId, 1, 2);
+        TestUtils.flushCurrentRQWithOldDeviceId(oldDeviceId); // clean current rq with old device id requests
+
+        deviceID.value += "1";
```

Comment:
still necessary?

Reply:
well It is a little bit complicated validation. Because it is atomic it needed that way but I can set it more readeble